analyzers/src/SonarAnalyzer.CSharp/Rules/BooleanLiteralUnnecessaryCodeFix.cs

```diff
@@ -223,10 +223,28 @@ private static SyntaxNode FindNodeToKeep(BinaryExpressionSyntax binary)
 
             #endregion
 
+            #region ==/!= one side boolean
+
+            if (binary.IsKind(SyntaxKind.EqualsExpression))
+            {
+                // edge case [condition == false] -> !condition
+                if (CSharpEquivalenceChecker.AreEquivalent(binary.Right, CSharpSyntaxHelper.FalseLiteralExpression))
+                {
+                    return SyntaxFactory.PrefixUnaryExpression(SyntaxKind.LogicalNotExpression, binary.Left);
+                }
+                // edge case [false == condition] -> !condition
+                if (CSharpEquivalenceChecker.AreEquivalent(binary.Left, CSharpSyntaxHelper.FalseLiteralExpression))
+                {
+                    return SyntaxFactory.PrefixUnaryExpression(SyntaxKind.LogicalNotExpression, binary.Right);
+                }
```

Comment:
There is some duplication here, because I am doing the same for left->right and right->left.
But I don't think it is big enough to extract.